helper/battery.c

```diff
@@ -49,48 +49,45 @@ const uint16_t 	  lowBatteryPeriod = 30;
 volatile uint16_t gPowerSave_10ms;
 
 
-unsigned int BATTERY_VoltsToPercent(const unsigned int voltage_10mV)
-{
-	const uint16_t crv1600[][2] = {
+const uint16_t Voltage2PercentageTable[][7][2] = {
+	[BATTERY_TYPE_1600_MAH] = {
 		{828, 100},
 		{814, 97 },
 		{760, 25 },
 		{729, 6  },
 		{630, 0  },
-		{0,   0  }
-	};
+		{0,   0  },
+		{0,   0  },
+	},
 
-	const uint16_t crv2200[][2] = {
+	[BATTERY_TYPE_2200_MAH] = {
 		{832, 100},
 		{813, 95 },
 		{740, 60 },
 		{707, 21 },
 		{682, 5  },
 		{630, 0  },
-		{0,   0  }
-	};
-	
-	const BATTERY_Type_t type = gEeprom.BATTERY_TYPE;
-	const uint16_t(*crv)[2];
-	uint8_t size;
-	if (type == BATTERY_TYPE_2200_MAH) {
-		crv = crv2200;
-		size = ARRAY_SIZE(crv2200);
-	}	
-	else {
-		crv = crv1600;
-		size = ARRAY_SIZE(crv1600);
-	}
+		{0,   0  },
+	},
+};
+
+static_assert(ARRAY_SIZE(Voltage2PercentageTable[BATTERY_TYPE_1600_MAH]) ==
```

Comment:
Lack of `#include <assert.h>` #264

Reply:
I just tested, this is not a windows thing, this happens in arm-none-eabi-gcc v10 which is used in windows, but the same error appears on WSL.

Reply:
I did fix that already, did you try with the include removed?
this is the actual output from my WSL
```
uv-k5-firmware-custom$ arm-none-eabi-gcc --version
arm-none-eabi-gcc (15:10.3-2021.07-4) 10.3.1 20210621 (release)
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```